docs/src/main/asciidoc/security-customization.adoc

```diff
@@ -0,0 +1,152 @@
+////
+This guide is maintained in the main Quarkus repository
+and pull requests should be submitted there:
+https://github.com/quarkusio/quarkus/tree/master/docs/src/main/asciidoc
+////
+= Quarkus - Security Tips and Tricks
+
+include::./attributes.adoc[]
+
+== HttpAuthenticationMechanism Customization
+
+One can customize `HttpAuthenticationMechanism` by registering an `ApplicationScoped` implementation bean.
```

Comment:
Why must it be an `ApplicationScoped` ? Isn't a `Singleton` working ?
For customization via CDI bean we usually just say something like `by providing a CDI bean that implements ...` then in the example uses `Singleton` if possible (no proxy generation) or `ApplicationScoped`.

Reply:
@loicmathieu I've really no idea, I've only seen `ApplicationScoped`. Would it work if I say an `ApplicationScoped or Singleton` ?

Reply:
In fact, better check with @stuartwdouglas if `@Singleton` will works here. If yes better to suggest people to use this scope.

Reply:
@loicmathieu Hi, yeah, I've just updated the line to  `One can customize HttpAuthenticationMechanism` by registering a CDI implementation bean.` to avoid mentioning `ApplicationScoped`, though it is still in the listed example.

Reply:
Good for me. It's a valid implementation so it should not be important

Reply:
@loicmathieu Thanks, if you'd like, please open an issue to investigate running authenticators as `Singleton`s, suppose would be good if it worked for the perf reasons